app/lib/irs1040_pdf.rb

```diff
@@ -1,92 +1,87 @@
+    @spouse_first_name = intake.spouse_middle_initial.present? ? "#{intake.spouse_first_name} #{intake.spouse_middle_initial}" : intake.spouse_first_name
+    @spouse_last_name = intake.spouse_last_name
+    address = submission.verified_address || intake.address
+    @street_address = address.street_address
+    @city = address.city
+    @state = address.state
+    @zip_code = address.zip_code
+
+    # Most PDF fields are grabbed right off the XML
+    @xml_document = SubmissionBuilder::Ty2021::Return1040.new(submission).document
   end
 
   def hash_for_pdf
-    total_refundable_credits = @benefits.claimed_recovery_rebate_credit.to_i + @benefits.outstanding_ctc_amount
     answers = {
-        FilingStatus: @tax_return.filing_status_code,
-        PrimaryFirstNm: @intake.primary_middle_initial.present? ? "#{@intake.primary_first_name} #{@intake.primary_middle_initial}" : @intake.primary_first_name,
-        PrimaryLastNm: @intake.primary_last_name,
-        PrimarySSN: @intake.primary_ssn,
-        AddressLine1Txt: @address&.street_address,
-        CityNm: @address&.city,
-        StateAbbreviationCd: @address&.state,
-        ZipCd: @address&.zip_code,
-        VirtualCurAcquiredDurTYInd: @intake.has_crypto_income,
-        PrimaryBlindInd: bool_checkbox(@intake.was_blind_yes?),
-        TotalItemizedOrStandardDedAmt12a: @tax_return.standard_deduction,
-        TotalAdjustmentsToIncomeAmt12c: @tax_return.standard_deduction, # 12c = 12a + 12b; 12b is charitable contributions which is 0 for us
-        TotalDeductionsAmt14: @tax_return.standard_deduction, # 14 = 12c + 13; 13 is business income deduction which we leave blank
-        TaxableIncomeAmt15: 0,
-        AdditionalChildTaxCreditAmt28: @benefits.outstanding_ctc_amount,
-        RecoveryRebateCreditAmt30: @benefits.claimed_recovery_rebate_credit.to_i,
-        RefundableCreditsAmt32: total_refundable_credits,
-        TotalPaymentsAmt33: total_refundable_credits,
-        OverpaidAmt34: total_refundable_credits,
-        RefundAmt35: total_refundable_credits,
-        Primary65OrOlderInd: bool_checkbox(@tax_return.primary_age_65_or_older?),
-        PrimarySignature: @intake.primary_full_name,
-        PrimarySignatureDate: @intake.primary_signature_pin_at&.strftime("%m/%d/%y"),
-        PrimaryIPPIN: @intake.primary_ip_pin,
-        PhoneNumber:  @intake.formatted_phone_number || @intake.formatted_sms_phone_number,
-        EmailAddress: @intake.email_address
+      FilingStatus: @xml_document.at("IndividualReturnFilingStatusCd")&.text,
+      PrimaryFirstNm: @primary_first_name,
+      PrimaryLastNm: @primary_last_name,
+      PrimarySSN: @xml_document.at("PrimarySSN")&.text,
+      AddressLine1Txt: @street_address,
+      CityNm: @city,
+      StateAbbreviationCd: @state,
+      ZipCd: @zip_code,
+      VirtualCurAcquiredDurTYInd: @xml_document.at("VirtualCurAcquiredDurTYInd")&.text,
+      PrimaryBlindInd: bool_from_xml(@xml_document.at("PrimaryBlindInd")) ? "1" : "Off",
```

Comment:
xml_check_to_bool() may make more sense as a method name here, since we're specifically looking for the presence of the X character. Also noodling on whether it would make sense to continue having a helper to create the "right" check vs Off value for the PDF vs doing a ternary in multiple places